File: cannlytics/models/dataset.py
"""
Data Set Model | Cannlytics
Copyright (c) 2021-2022 Cannlytics and Cannlytics Contributors

Authors: Keegan Skeate <keegan@cannlytics.com>
Created: 11/5/2021
Updated: 11/5/2021
License: <https://github.com/cannlytics/cannlytics-engine/blob/main/LICENSE>

Description: Data set data model.
"""
# Standard imports.
from dataclasses import dataclass

# Internal imports.
from .base import Model


@dataclass
class DataSet(Model):
    """An abstract representation of a data set."""
    _collection = 'organizations/%s/datasets'

File: data/lib/models/dataset.dart
// Cannlytics Data
// Copyright (c) 2023 Cannlytics

// Authors:
//   Keegan Skeate <https://github.com/keeganskeate>
// Created: 4/15/2023
// Updated: 5/14/2023
// License: MIT License <https://github.com/cannlytics/cannlytics/blob/main/LICENSE>

// Flutter imports:
import 'package:flutter/material.dart';

/// Model representing a dataset.
class Dataset {
  final String? svgSrc;
  final String? title;
  final String? totalStorage;
  final int? numOfFiles;
  final int? percentage;
  final Color? color;

  final String? name;
  final String? description;
  final Size? size;
  final List? observations;
  final List<dynamic>? attributes;
  final String? source;
  final DateTime? lastUpdated;

  Dataset({
    this.svgSrc,
    this.title,
    this.totalStorage,
    this.numOfFiles,
    this.percentage,
    this.color,
    this.name,
    this.description,
    this.size,
    this.observations,
    this.attributes,
    this.source,
    this.lastUpdated,
  });

  factory Dataset.fromJson(Map<String, dynamic> json) {
    return Dataset(
      svgSrc: null,
      title: null,
      totalStorage: null,
      numOfFiles: null,
      percentage: null,
      color: null,
      name: json['dataset']['name'],
      description: json['dataset']['description'],
      size: json['dataset']['size'],
      observations: json['dataset']['observations'],
      attributes: json['dataset']['attributes'],
      source: json['dataset']['source'],
      lastUpdated: DateTime.parse(json['dataset']['last_updated']),
    );
  }
}
